language/e2e-tests/src/tests/transaction_builder.rs

```diff
@@ -65,6 +67,47 @@ fn register_preburn_burn() {
     ));
 }
 
+#[test]
+fn freeze_unfreeze_account() {
+    // create a FakeExecutor with a genesis from file
+    let mut executor = FakeExecutor::from_genesis_file();
+    // association account to do the actual burning
+    let association = Account::new_association();
+
+    let account = {
+        let data = AccountData::new(1, 1);
+        executor.add_account_data(&data);
+        data.into_account()
+    };
+
+    let lbr_ty = TypeTag::Struct(StructTag {
+        address: account_config::CORE_CODE_ADDRESS,
+        module: Identifier::new("LibraAccount").unwrap(),
+        name: Identifier::new("FreezingPrivilege").unwrap(),
+        type_params: vec![],
+    });
+
+    // Request freezing privilege
+    executor.execute_and_apply(
+        association.signed_script_txn(encode_apply_for_association_privilege(lbr_ty.clone()), 1),
+    );
+
+    // Grant freezing privilege to assocation account
+    executor.execute_and_apply(association.signed_script_txn(
+        encode_grant_association_privilege(lbr_ty, *association.address()),
+        2,
+    ));
+    // Execute freeze on account
+    executor.execute_and_apply(
+        association.signed_script_txn(encode_freeze_account(*account.address()), 3),
+    );
+
```

Comment:
Not strictly necessary, but could be useful to test that sending a tx from `account` fails after freeze and succeeds after unfreeze.

Reply:
Any transaction will do for testing this, but key rotation is probably the easiest: https://github.com/libra/libra/blob/master/language/e2e-tests/src/tests/rotate_key.rs